src/ray/common/task/task_spec.cc

```diff
@@ -215,6 +215,23 @@ ObjectID TaskSpecification::ReturnId(size_t return_index) const {
   return ObjectID::FromIndex(TaskId(), return_index + 1);
 }
 
+size_t TaskSpecification::NumStreamingGeneratorReturns() const {
+  return message_->num_streaming_generator_returns();
+}
+
+ObjectID TaskSpecification::StreamingGeneratorReturnId(size_t generator_index) const {
+  // Streaming generator task has only 1 return ID.
+  RAY_CHECK_EQ(NumReturns(), 1);
+  RAY_CHECK(generator_index < 100 * 1000 * 1000);
+  // 2 for a single return from a generator task.
```

Comment:
what's is 2?

Reply:
1 is reserved for the first task return.

Reply:
I can improve the doc. 

```
index 1 is reserved for the first task return from a generator task. 
```